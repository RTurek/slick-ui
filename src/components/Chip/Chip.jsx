import React from 'react';
import PropTypes from 'prop-types';
import { useTheme } from '@material-ui/core/styles';
import MuiChip from '@material-ui/core/Chip';

// https://mui.com/components/chips/

function Chip(props) {
  const { color, ...passedProps } = props;
  const theme = useTheme();
  const style = {
    ...props.style
  };
  const defaultVariants = ['primary', 'secondary'];
  if (!defaultVariants.includes(color)) {
    style.backgroundColor = theme.palette[props.color].main;
    return <MuiChip {...passedProps} style={style} />;
  }
  return <MuiChip {...passedProps} style={style} color={color} />;
}

Chip.propTypes = {
  /** Avatar Element */
  avatar: PropTypes.node,
  /** Class overrides */
  classes: PropTypes.shape({}),
  /** Appears to be interactive on Click */
  clickable: PropTypes.bool,
  /** Color, primary or secondary */
  color: PropTypes.oneOf([
    'primary',
    'secondary',
    'warning',
    'lightGrey',
    'mediumGrey',
    'alert',
    'success'
  ]),
  /** String or Node to be rendered inside component */
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  /** Function to fire when delete is clicked */
  onDelete: PropTypes.func,
  /** Varaint, default or outlined */
  variant: PropTypes.oneOf(['default', 'outlined']),
  /** optional override for style */
  style: PropTypes.shape({})
};

Chip.defaultProps = {
  avatar: null,
  classes: {},
  clickable: true,
  color: 'primary',
  onDelete: undefined,
  variant: 'default',
  style: undefined
};

export default Chip;
