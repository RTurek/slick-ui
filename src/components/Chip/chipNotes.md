#### Chip
---

This is a wrapper component for Material Chip, mostly to apply custom styling and to force only a subset of variant choices

#### Notes
---
This could be used as an input, or for displaying data in a more noticable way.

#### Sources
---
- https://mui.com/components/chips/
