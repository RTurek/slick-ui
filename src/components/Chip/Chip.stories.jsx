import React from 'react';
import { withInfo } from '@storybook/addon-info';
import { text, select, boolean } from '@storybook/addon-knobs';
import Chip from './Chip';

const colorOptions = {
  primary: 'primary',
  secondary: 'secondary'
};

export default {
  title: '1 ~ Slick / Uncategorized / Chip',

  decorators: [
    withInfo({
      text: '',
      header: true,
      inline: true,
      source: false,
      propTables: [Chip],
      excludedPropTypes: ['theme']
    })
  ]
};

export const containedDefault = () => (
  <Chip
    color={select('Color', colorOptions, 'primary')}
    label={text('Label', 'test@turekworks.com')}
    onDelete={boolean('Has Delete', true)}
    clickable={boolean('Clickable', true)}
  />
);

containedDefault.story = {
  name: 'Contained/Default'
};

export const outlined = () => (
  <Chip
    variant="outlined"
    color={select('Color', colorOptions, 'primary')}
    label={text('Label', 'test@turekworks.com')}
    onDelete={boolean('Has Delete', true)}
    clickable={boolean('Clickable', true)}
  />
);

outlined.story = {
  name: 'Outlined'
};
