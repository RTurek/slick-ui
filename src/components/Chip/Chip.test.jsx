import React from 'react';
import renderer from 'react-test-renderer';
// import ThemeWrapper from '@style/ThemeWrapper';
import Chip from './Chip';

test('Chip should render', () => {
  const component = (
    // <ThemeWrapper>
    <Chip label="test" />
    // </ThemeWrapper>
  );
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});
