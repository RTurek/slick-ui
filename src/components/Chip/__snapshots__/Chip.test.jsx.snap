// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip should render 1`] = `
<div
  className="MuiButtonBase-root MuiChip-root MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-clickable"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  style={Object {}}
  tabIndex={0}
>
  <span
    className="MuiChip-label"
  >
    test
  </span>
</div>
`;
